tests/acceptance/features/apiGraph/enforcePasswordPublicLink.feature

```diff
@@ -0,0 +1,35 @@
+@api @env-config
+Feature: enforce password on public link
+  As a user
+  I want to enforce passwords on public links shared with upload, edit, or contribute permission
+  So that the password is required to access the contents in the link
+
+  Background:
+    Given the config "OCIS_SHARING_PUBLIC_WRITEABLE_SHARE_MUST_HAVE_PASSWORD" has been set to "true"
+    And these users have been created with default attributes and without skeleton files:
+      | username |
+      | Alice    |
+    And user "Alice" has uploaded file with content "test file" to "/testfile.txt"
+
+
+  Scenario: user tries to update a public link to edit permission without a password when enforce-password is enabled
```

Comment:
I don't think the `when enforce-password is enabled` is needed because the whole feature about that
```suggestion
  Scenario: user tries to update a public link to edit permission without a password 
```